### PR TITLE
Add SemVerHelper.isValidSemVer(string)

### PR DESCRIPTION
**Pull request by nathanaeljones**
Allows build scripts to check if a git tag or input string is a semver.

### CODE REVIEW

File: src/app/FakeLib/SemVerHelper.fs [hunk 1]
```diff
@@ -76,6 +76,15 @@ type SemVerInfo =
                     0
             | _ -> invalidArg "yobj" "cannot compare values of different types"
 
+
+let private SemVerPattern = "^(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)(?:-[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?(?:\+[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?$"
```
Comment (forki) [inline src/app/FakeLib/SemVerHelper.fs]: shouldn't this be part of our normal Semver parser?

Comment (nathanaeljones) [inline src/app/FakeLib/SemVerHelper.fs]: You're saying we should throw an exception during `parse` for parsable but technically invalid semver strings? Int32.Parse doesn't align with the spec, particularly if we think about culture. While we can improve parse (and possibly make it more strict), some things (like prohibiting leading zeroes) are more complex without a regex.

Comment (forki) [inline src/app/FakeLib/SemVerHelper.fs]: Ok let's ask differently. Why do you need this new function? What's the use
case?

Comment (nathanaeljones) [inline src/app/FakeLib/SemVerHelper.fs]: I use it to determine if a git tag is a semver. If it's a semver, then the build can take additional actions. Very useful for continuous delivery.

**Pull request merged**